### PR TITLE
Do not run 'remove unnecessary imports' on generated code

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/74761.

1. this can be very expensive.
2. it's unlikely the user can do anything about this.  the code is generated, so it's not like they can remove the imports from teh generated code.

### CODE REVIEW

File: src/Analyzers/CSharp/Tests/RemoveUnnecessaryImports/RemoveUnnecessaryImportsTests.cs [hunk 1]
```diff
@@ -186,22 +186,9 @@ public async Task TestGeneratedCode()
         var source = """
```
Comment (DoctorKrolic) [inline src/Analyzers/CSharp/Tests/RemoveUnnecessaryImports/RemoveUnnecessaryImportsTests.cs:186]: Maybe add a `[WorkItem]` to the test to clarify, why no diagnostics there are expected

**Review by Youssef1313 — APPROVED** [review]
LGTM, but one thing to note. Do we know for sure that this is actually a performance issue? In some cases, the analyzer can be showing as expensive because it's calculating something that will happen anyways during the compilation (i.e, the cost is just moving around between the rest of the compilation process and the analyzer).

**Review by sharwell — DISMISSED** [review]
This is a red herring. Remove unnecessary imports has no actual performance impact. It _incorrectly_ captures unavoidable performance overhead from a different part of the system, suggesting it can be removed by disabling this analyzer when in fact it cannot.

**Comment by CyrusNajmabadi** [comment]
@sharwell what value does this analyzer have though?  The user cannot change the usings added. we don't run other style analyzers on generated code, why do we run this one. 

> Remove unnecessary imports has no actual performance impact. 

Then we shouldn't report that it does. It also makes the other analyzers look good by comparison. 

I'd still like to take this though, if there's no use value provided.

**Comment by CyrusNajmabadi** [comment]
The best code is no code.  I don't see why we are special casing this style analyzer for generated code, and nothing else.

**Review by akhera99 — APPROVED** [review]
looks good, I agree that the test should have some indication that the behavior has changed

**Pull request merged**